src/Symfony/Component/Cache/Traits/RedisTrait.php

```
@@ -113,54 +114,58 @@ public static function createConnection($dsn, array $options = [])
             return 'file:'.($m[1] ?? '');
         }, $dsn);
 
-        if (false === $params = parse_url($params)) {
+        if (false === $parsedUrl = parse_url($url)) {
             throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s".', $dsn));
         }
 
-        $query = $hosts = [];
+        $params = [];
 
-        if (isset($params['query'])) {
-            parse_str($params['query'], $query);
+        if (isset($parsedUrl['path']) && preg_match('#/(\d+)$#', $parsedUrl['path'], $m)) {
+            $params['dbindex'] = $m[1];
+            $parsedUrl['path'] = substr($parsedUrl['path'], 0, -\strlen($m[0]));
+        }
+
+        $query = [];
+        if (isset($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $query);
+        }
+
+        $params += $query + $options + self::$defaultConnectionOptions;
+
+        $tls = 'rediss' === $dsnScheme || filter_var($params['tls'], \FILTER_VALIDATE_BOOLEAN);
```

`rediss://host?tls=false` does not make sens and should throw an exception.

---

`$params['tls']` may not exists